client/opentracing/src/test/java/io/opentracing/AsyncService.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentracing;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMapExtractAdapter;
+import io.opentracing.propagation.TextMapInjectAdapter;
+
+/**
+ * @author gbrown
+ */
+public class AsyncService {
+
+    private Tracer tracer;
+
+    private Message outboundMessage;
+
+    public AsyncService(Tracer tracer) {
+        this.tracer = tracer;
+    }
+
+    public void handle(Message message) {
+        SpanContext spanCtx = tracer.extract(Format.Builtin.TEXT_MAP,
+                new TextMapExtractAdapter(message.getHeaders()));
+
+        // Top level, so create Tracer and root span
+        Span serverSpan = tracer.buildSpan("Server")
+                .asChildOf(spanCtx)
+                .withTag("http.url", "http://localhost:8080/inbound?orderId=123&verbose=true")
+                .withTag("orderId", "1243343456455")
+                .start();
+
+        try {
+            synchronized (this) {
+                wait(500);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        component1(serverSpan);
+
+        serverSpan.finish();
+
+        serverSpan.close();
+    }
+
+    public void component1(Span span) {
+        // Span will auto-close and automatically call finish
+        try (Span component1Span = tracer.buildSpan("Component1")
+                .asChildOf(span)
+                .start()) {
+
+            try {
+                synchronized (this) {
+                    wait(500);
```

Comment:
I see lot of these `wait(500)` wouldn't be better to extract it to a helper method?